src/platform/Tizen/WiFiManager.cpp

```diff
@@ -219,6 +344,98 @@ void WiFiManager::_ScanFinishedCb(wifi_manager_error_e wifiErr, void * userData)
     }
 }
 
+void WiFiManager::_ScanFinishedCb(wifi_manager_error_e wifiErr, void * userData)
+{
+    ChipLogProgress(DeviceLayer, "ScanFinished Callback..");
+
+    if (wifiErr == WIFI_MANAGER_ERROR_NONE)
+    {
+        ChipLogProgress(DeviceLayer, "WiFi scan finished");
+        std::vector<WiFiScanResponse> * networkScanned = new std::vector<WiFiScanResponse>();
+
+        int err = wifi_manager_foreach_found_specific_ap(sInstance.mWiFiManagerHandle, _FoundAPOnScanCb, networkScanned);
+        if (err == WIFI_MANAGER_ERROR_NONE)
+        {
+            if (sInstance.mpScanCallback != nullptr)
+            {
+                TizenScanResponseIterator<WiFiScanResponse> iter(const_cast<std::vector<WiFiScanResponse> *>(networkScanned));
+                chip::DeviceLayer::PlatformMgr().LockChipStack();
+                sInstance.mpScanCallback->OnFinished(Status::kSuccess, CharSpan(), &iter);
+                chip::DeviceLayer::PlatformMgr().UnlockChipStack();
+                sInstance.mpScanCallback = nullptr;
+            }
```

Comment:
Since the ` sInstance.mpScanCallback` is set in the main Matter thread in the `StartWiFiScan` function, I would go for checking and setting this variable under chip stack lock:

```suggestion
            chip::DeviceLayer::PlatformMgr().LockChipStack();
            if (sInstance.mpScanCallback != nullptr)
            {
                TizenScanResponseIterator<WiFiScanResponse> iter(const_cast<std::vector<WiFiScanResponse> *>(networkScanned));
                sInstance.mpScanCallback->OnFinished(Status::kSuccess, CharSpan(), &iter);
                sInstance.mpScanCallback = nullptr;
            }
            chip::DeviceLayer::PlatformMgr().UnlockChipStack();
```